### PR TITLE
Adds the DeviceLinker mapping tool

### PR DESCRIPTION
**Pull request by NoooneyDude**
- Adds the `Device Linker` mapping tool.
device linker tool
- Small code tweaks, namespacing.
- CL: [Balance] APC connections are now limited by distance: you can no longer power a whole asteroid base from a single APC on the main station.
Only applies to future map changes or changes during gameplay.

### CODE REVIEW

**Comment by NoooneyDude** [comment]
The Codacy complaints must be ignored.

**Comment by PerfectTangent** [comment]
is there an easy way to visualize connections in the maps that are too far so we can clean up maps?

**Comment by NoooneyDude** [comment]
Yes, hit the `Next` button. It'll focus on the object in question.

~Edit: oh, well, there's no gizmo to show all problematic devices at once, no.~ ACUs, APCs show a disconnected symbol if gizmos are turned on.

**Comment by NoooneyDude** [comment]
If I'm being honest, 30 tiles is quite a distance. Could probably drop that to 25 or perhaps even 20 and still manage to reach all devices in atmospherics and have reasonable spacing along halls.

**Comment by corp-0** [comment]
code looks good. The tool is very neat looking. Update the branch so we can approve and merge.

File: UnityProject/Assets/Scripts/Core/Editor/Tools/Mapping/DeviceLinker.cs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+				SceneView.FrameLastActiveSceneView();
+			}
+
+			if (Tab.ReviewDistantSlaveIndex > -1) 
+			{
+				var slaveUnderReview = Tab.DistantSlaves[Tab.ReviewDistantSlaveIndex];
+				float distance = DeviceLinker.LinkSlave(slaveUnderReview);
+				if (distance < DeviceLinker.Masters[0].MaxDistance)
+				{
+					GUILayout.Label($"<b>{slaveUnderReview.gameObject.name}</b> has been relinked.</b>");// TODO: it has been relinked but what if it hasn"t does this even make sense halp me
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Editor/Tools/Mapping/DeviceLinker.cs]:
```suggestion
					GUILayout.Label($"<b>{slaveUnderReview.gameObject.name}</b> has been relinked.</b>");
```

**Review by Bod9001 — APPROVED** [review]
Looks good

**Comment by Bod9001** [comment]
man, this is going to take forever to Merge things

**Review by ThatDan123 — APPROVED** [review]

**Pull request merged**